pkg/kapis/devops/v1alpha3/scm/scmhandler.go

```diff
@@ -48,14 +55,82 @@ func (h *handler) verify(request *restful.Request, response *restful.Response) {
 	secretNamespace := request.QueryParameter("secretNamespace")
 	server := common.GetQueryParameter(request, queryParameterServer)
 
-	_, code, err := h.getOrganizations(scm, server, secretName, secretNamespace, 1, 1, false)
+	code, err := 0, error(nil)
+	switch scm {
+	case "git":
+		if server == "" {
+			err := fmt.Errorf("server is required")
+			kapis.HandleError(request, response, err)
+			response.WriteHeaderAndEntity(http.StatusBadRequest, err)
+			return
+		}
+		code, err = h.checkRepoAccess(server, secretName, secretNamespace)
+
+	default:
+		_, code, err = h.getOrganizations(scm, server, secretName, secretNamespace, 1, 1, false)
+	}
 
 	response.Header().Set(restful.HEADER_ContentType, restful.MIME_JSON)
 	verifyResult := git.VerifyResult(err, code)
 	verifyResult.CredentialID = secretName
 	_ = response.WriteAsJson(verifyResult)
 }
 
+func (h *handler) checkRepoAccess(repourl, secretName, secretNamespace string) (int, error) {
+	user, token, privateKey, err := h.getTokenWithSShFromSecret(
```

Comment:
when using `git` to import a repo, a secret is optional. e.g. a public repo. It seems you did not handle this situation.

Reply:
done